File: backend/schemas/meta.py
from pydantic import BaseModel
from typing import List, Optional


class ColumnMeta(BaseModel):
    name: str
    type: str


class RelationshipMeta(BaseModel):
    relation: str
    related_model: str
    foreign_keys: List[str]


class TableMeta(BaseModel):
    table_name: str
    columns: List[ColumnMeta]
    relationships: List[RelationshipMeta]

File: frontend/src/domain/interfaces/meta.ts
export interface IColumnMeta {
  name: string;
  type: string;
  is_rel?: boolean;
}

export interface IRelationshipMeta {
  relation: string;
  related_model: string;
  foreign_keys: string[];
}

export interface ITableMeta {
  table_name: string;
  columns: IColumnMeta[];
  relationships: IRelationshipMeta[];
}

export interface IItemMeta {
  name: string;
  description: string;
}
